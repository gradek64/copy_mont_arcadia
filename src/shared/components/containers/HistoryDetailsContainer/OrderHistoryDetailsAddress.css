/** @define OrderHistoryDetailsAddress */

.OrderHistoryDetailsAddress {
  padding: 10px $page-padding 15px;
  border-top: 1px solid $myaccount-border-color;

  &-title {
    margin-top: 10px;
  }

  &-body {
    display: flex;
  }

  &-icon {
    padding: 10px;
    margin-right: 15px;
    width: 50px;
    align-self: flex-start;
  }
  &-label {
    margin-bottom: 10px;
  }

  &-details {
    margin: 0;
    padding: 10px 0;
  }

  &-line {
    display: block;
  }
}

@media $min-tablet {
  .OrderHistoryDetailsAddress {
    border-top: none;

    &-title {
      border-bottom: 1px solid $myaccount-divider-color;
      padding-bottom: 8px;
    }
  }
}
