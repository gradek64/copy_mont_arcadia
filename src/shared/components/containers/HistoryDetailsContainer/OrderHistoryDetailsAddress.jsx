import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { PrivacyGuard } from '../../../lib'

import Image from '../../common/Image/Image'

export default class OrderHistoryDetailsAddress extends Component {
  static propTypes = {
    type: PropTypes.string.isRequired,
    deliveryMethod: PropTypes.string,
    address: PropTypes.object,
    className: PropTypes.string,
  }

  static contextTypes = {
    l: PropTypes.func,
  }

  static defaultProps = {
    className: '',
  }

  getAddressIcon = (type, deliveryMethod = '') => {
    const homeIcon = '/assets/{brandName}/images/checkout-home.svg'
    const storeIcon = '/assets/{brandName}/images/arcadia-store-icon.svg'

    if (type !== 'billing' && deliveryMethod.includes('Collect'))
      return storeIcon
    return homeIcon
  }

  render() {
    const { l } = this.context
    const { address, type, deliveryMethod, className } = this.props
    const title = type === 'billing' ? l`Payment details` : l`Delivery Details`

    const addressIcon = this.getAddressIcon(type, deliveryMethod)

    if (!address) return null
    return (
      <section className={`OrderHistoryDetailsAddress ${className}`}>
        <h3 className="OrderHistoryDetailsAddress-title">{title}</h3>
        <div className="OrderHistoryDetailsAddress-body">
          <Image
            className="OrderHistoryDetailsAddress-icon"
            src={addressIcon}
          />
          <p className="OrderHistoryDetailsAddress-details">
            {type === 'billing' && (
              <span className="OrderHistoryDetailsAddress-line OrderHistoryDetailsAddress-label">
                {l`Billing address`}:
              </span>
            )}
            {address.name && (
              <PrivacyGuard>
                <span className="OrderHistoryDetailsAddress-line">
                  {address.name}
                </span>
              </PrivacyGuard>
            )}
            {address.address1 && (
              <PrivacyGuard>
                <span className="OrderHistoryDetailsAddress-line">
                  {address.address1}
                </span>
              </PrivacyGuard>
            )}
            {address.address2 && (
              <PrivacyGuard>
                <span className="OrderHistoryDetailsAddress-line">
                  {address.address2}
                </span>
              </PrivacyGuard>
            )}
            {address.address3 && (
              <PrivacyGuard>
                <span className="OrderHistoryDetailsAddress-line">
                  {address.address3}
                </span>
              </PrivacyGuard>
            )}
            {address.address4 && (
              <PrivacyGuard>
                <span className="OrderHistoryDetailsAddress-line">
                  {address.address4}
                </span>
              </PrivacyGuard>
            )}
            {address.country && (
              <PrivacyGuard>
                <span className="OrderHistoryDetailsAddress-line">
                  {address.country}
                </span>
              </PrivacyGuard>
            )}
          </p>
        </div>
      </section>
    )
  }
}
