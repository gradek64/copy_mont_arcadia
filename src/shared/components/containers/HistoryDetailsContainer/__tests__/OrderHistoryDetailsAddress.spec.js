import testComponentHelper from 'test/unit/helpers/test-component'
import OrderHistoryDetailsAddress from '../OrderHistoryDetailsAddress'

describe('<OrderHistoryDetailsAddress />', () => {
  const renderComponent = testComponentHelper(OrderHistoryDetailsAddress)
  const address = {
    name: 'Miss Kate',
    address1: 'address1',
    address2: 'address2',
    address3: 'address3',
    address4: 'address4',
    country: 'UK',
  }
  const initialProps = {
    address,
    type: 'shipping',
  }

  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(initialProps).getTree()).toMatchSnapshot()
    })
    it('with className', () => {
      expect(
        renderComponent({
          ...initialProps,
          className: 'OrderHistoryDetailsAddress--delivery',
        }).getTree()
      ).toMatchSnapshot()
    })
    it('with no address', () => {
      expect(
        renderComponent({
          ...initialProps,
          address: undefined,
        }).getTree()
      ).toMatchSnapshot()
    })
    it('with address and type shipping and deliveryMethod is "Collect from store" ', () => {
      expect(
        renderComponent({
          ...initialProps,
          address,
          deliveryMethod: 'Collect from store',
        }).getTree()
      ).toMatchSnapshot()
    })
    it('with address and type billing', () => {
      expect(
        renderComponent({
          ...initialProps,
          address,
          type: 'billing',
        }).getTree()
      ).toMatchSnapshot()
    })
  })
})
