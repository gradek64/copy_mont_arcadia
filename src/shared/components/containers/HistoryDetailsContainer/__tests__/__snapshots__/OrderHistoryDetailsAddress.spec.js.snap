// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderHistoryDetailsAddress /> @renders in default state 1`] = `
<section
  className="OrderHistoryDetailsAddress "
>
  <h3
    className="OrderHistoryDetailsAddress-title"
  >
    Delivery Details
  </h3>
  <div
    className="OrderHistoryDetailsAddress-body"
  >
    <Connect(Image)
      className="OrderHistoryDetailsAddress-icon"
      src="/assets/{brandName}/images/checkout-home.svg"
    />
    <p
      className="OrderHistoryDetailsAddress-details"
    >
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          Miss Kate
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          address1
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          address2
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          address3
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          address4
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          UK
        </span>
      </PrivacyGuard>
    </p>
  </div>
</section>
`;

exports[`<OrderHistoryDetailsAddress /> @renders with address and type billing 1`] = `
<section
  className="OrderHistoryDetailsAddress "
>
  <h3
    className="OrderHistoryDetailsAddress-title"
  >
    Payment details
  </h3>
  <div
    className="OrderHistoryDetailsAddress-body"
  >
    <Connect(Image)
      className="OrderHistoryDetailsAddress-icon"
      src="/assets/{brandName}/images/checkout-home.svg"
    />
    <p
      className="OrderHistoryDetailsAddress-details"
    >
      <span
        className="OrderHistoryDetailsAddress-line OrderHistoryDetailsAddress-label"
      >
        Billing address
        :
      </span>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          Miss Kate
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          address1
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          address2
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          address3
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          address4
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          UK
        </span>
      </PrivacyGuard>
    </p>
  </div>
</section>
`;

exports[`<OrderHistoryDetailsAddress /> @renders with address and type shipping and deliveryMethod is "Collect from store"  1`] = `
<section
  className="OrderHistoryDetailsAddress "
>
  <h3
    className="OrderHistoryDetailsAddress-title"
  >
    Delivery Details
  </h3>
  <div
    className="OrderHistoryDetailsAddress-body"
  >
    <Connect(Image)
      className="OrderHistoryDetailsAddress-icon"
      src="/assets/{brandName}/images/arcadia-store-icon.svg"
    />
    <p
      className="OrderHistoryDetailsAddress-details"
    >
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          Miss Kate
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          address1
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          address2
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          address3
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          address4
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          UK
        </span>
      </PrivacyGuard>
    </p>
  </div>
</section>
`;

exports[`<OrderHistoryDetailsAddress /> @renders with className 1`] = `
<section
  className="OrderHistoryDetailsAddress OrderHistoryDetailsAddress--delivery"
>
  <h3
    className="OrderHistoryDetailsAddress-title"
  >
    Delivery Details
  </h3>
  <div
    className="OrderHistoryDetailsAddress-body"
  >
    <Connect(Image)
      className="OrderHistoryDetailsAddress-icon"
      src="/assets/{brandName}/images/checkout-home.svg"
    />
    <p
      className="OrderHistoryDetailsAddress-details"
    >
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          Miss Kate
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          address1
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          address2
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          address3
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          address4
        </span>
      </PrivacyGuard>
      <PrivacyGuard>
        <span
          className="OrderHistoryDetailsAddress-line"
        >
          UK
        </span>
      </PrivacyGuard>
    </p>
  </div>
</section>
`;

exports[`<OrderHistoryDetailsAddress /> @renders with no address 1`] = `""`;
